model/common/src/icon4py/model/common/orchestration/decorator.py

```diff
@@ -539,35 +537,37 @@ def dace_specific_kwargs(
 
         The additional kwargs are the connectivity tables (runtime tables) and the GHEX C++ pointers.
         """
-        return {
+        dace_kwargs = {
             # connectivity tables at runtime
-            **{
-                connectivity_identifier(k): v.table
-                for k, v in offset_providers.items()
-                if hasattr(v, "table")
-            },
-            # GHEX C++ ptrs
-            "__context_ptr": expose_cpp_ptr(exchange_obj._context)
-            if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
-            else 0,
-            "__comm_ptr": expose_cpp_ptr(exchange_obj._comm)
-            if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
-            else 0,
-            **{
-                f"__pattern_{dim.value}Dim_ptr": expose_cpp_ptr(exchange_obj._patterns[dim])
+            connectivity_identifier(k): v.table
+            for k, v in offset_providers.items()
+            if hasattr(v, "table")
+        }
+        if exchange_obj:
```

Comment:
Would it be better to set all these keys to the default value (0)?

Reply:
I would favour the `if exchange_obj is not None and not isinstance(exchange_obj, decomposition.SingleNodeExchange)` and set the pointers to 0.